nat/service_iptables.go

```diff
@@ -36,115 +34,87 @@ func (service *serviceIPTables) Add(rule RuleForwarding) {
-	out, err := exec.Command("sysctl", "-n", "net.ipv4.ip_forward").CombinedOutput()
+func (service *serviceIPTables) isIPForwardingEnabled() (enabled bool) {
+	cmd := utils.SplitCommand("/sbin/sysctl", "-n net.ipv4.ip_forward")
+	output, err := cmd.Output()
 	if err != nil {
-		return false, fmt.Errorf("Failed to check IP forwarding status: %s", err)
+		log.Warn("Failed to check IP forwarding status: ", cmd.Args, " Returned exit error: ", err.Error(), " Cmd output: ", string(output))
 	}
 
-	if strings.TrimSpace(string(out)) == "1" {
-		log.Info(NatLogPrefix, "IP forwarding already enabled")
-		return true, nil
+	if strings.TrimSpace(string(output)) == "1" {
+		log.Info(natLogPrefix, "IP forwarding already enabled")
+		return true
 	}
-	return false, nil
+	return false
 }
 
-func (service *serviceIPTables) enableIPForwarding() (err error) {
-
-	enabled, err := service.isIPForwardingEnabled()
-	if err != nil {
-		return err
-	}
+func (service *serviceIPTables) enableIPForwarding() {
+	enabled := service.isIPForwardingEnabled()
 
 	if enabled {
 		service.forward = true
-		return nil
+		return
 	}
-
-	cmd := exec.Command("sysctl", "-w", "net.ipv4.ip_forward=1")
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("Failed to enable IP forwarding: %s", err)
+	cmd := utils.SplitCommand("sudo", "/sbin/sysctl -w net.ipv4.ip_forward=1")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		log.Warn("Failed to enable IP forwarding: ", cmd.Args, " Returned exit error: ", err.Error(), " Cmd output: ", string(output))
+	} else {
+		log.Info(natLogPrefix, "IP forwarding enabled")
 	}
-
-	log.Info(NatLogPrefix, "IP forwarding enabled")
-	return nil
 }
 
-func (service *serviceIPTables) disableIPForwarding() (err error) {
+func (service *serviceIPTables) disableIPForwarding() {
 	if service.forward {
-		return nil
+		return
 	}
 
-	cmd := exec.Command("sysctl", "-w", "net.ipv4.ip_forward=0")
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("Failed to disable IP forwarding. %s", err)
+	cmd := utils.SplitCommand("sudo", "/sbin/sysctl -w net.ipv4.ip_forward=0")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		log.Warn("Failed to disable IP forwarding: ", cmd.Args, " Returned exit error: ", err.Error(), " Cmd output: ", string(output))
+	} else {
+		log.Info(natLogPrefix, "IP forwarding disabled")
 	}
-
-	log.Info(NatLogPrefix, "IP forwarding disabled")
-	return nil
 }
 
-func (service *serviceIPTables) enableRules() error {
+func (service *serviceIPTables) enableRules() {
 	for _, rule := range service.rules {
-		cmd := exec.Command(
-			"iptables",
-			"--table", "nat",
-			"--append", "POSTROUTING",
-			"--source", rule.SourceAddress,
-			"!", "--destination", rule.SourceAddress,
-			"--jump", "SNAT",
-			"--to", rule.TargetIP,
-		)
-		if err := cmd.Run(); err != nil {
-			return fmt.Errorf("Failed to create ip forwarding rule: %s. %s", cmd.Args, err.Error())
+		arguments := "/sbin/iptables --table nat --append POSTROUTING --source " +
```

Comment:
I would split dynamic arguments
- creates more possibilities to inject (your function is not tested)
- you have to concatenate then as strings anyway